back/engines/commercial/analysis/app/lib/analysis/auto_tagging_method/platform_topic.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Analysis
+  class AutoTaggingMethod::PlatformTopic < AutoTaggingMethod::Base
+    TAG_TYPE = 'platform_topic'
+
+    protected
+
+    # Use `execute` on the parent class to actually use the method
+    def run
+      tags_by_name = analysis.tags.where(tag_type: TAG_TYPE).all.group_by(&:name)
+      multiloc_service = MultilocService.new
+
+      IdeasTopic
+        .where(idea: analysis.inputs)
+        .includes(:topic)
+        .each do |idea_topic|
+          tag_name = multiloc_service.t(idea_topic.topic.title_multiloc)
+          if tags_by_name[tag_name].blank?
```

Comment:
Taking this along for next iteration(s), I doubted about this, also about adding a `code` attribute to the tags for 'fixed' tags for e.g. sentiment. We'll have to work on dealing with updated content soon, and then this will become more relevant.